### PR TITLE
Adding support to Priam for running local instances

### PR DESCRIPTION
**Pull request by ToadJam**
Adding support to Priam for running local instances. Includes a new YAML configuration specific to local builds and updated README documentation.

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java [hunk 1]
```diff
@@ -29,6 +29,9 @@
     private String cassHome;
 
     @JsonProperty
+    private String yamlLocation;
```
Comment (fahdsiddiqui) [inline priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java]: `@JsonIgnore` for this property. So, we can use the default if yamlLocation is not provided (in production system)

File: priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java [hunk 2]
```diff
@@ -183,7 +186,7 @@ public String getCassHome() {
     }
 
     public String getYamlLocation() {
-        return cassHome + "/conf/cassandra.yaml";
+        return yamlLocation;
```
Comment (fahdsiddiqui) [inline priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java]: `return Objects.firstNonNull(yamlLocation, cassHome + "/conf/cassandra.yaml");` ?

File: priam/src/main/java/com/netflix/priam/config/PriamConfiguration.java [hunk 3]
```diff
@@ -47,6 +47,9 @@
     @JsonProperty
     private int jvmMutexPort = 8086;
 
+    @JsonProperty
```
Comment (fahdsiddiqui) [inline priam/src/main/java/com/netflix/priam/config/PriamConfiguration.java]: `@JsonIgnore`

File: priam/src/main/resources/conf/priam.yaml [hunk 4]
```diff
@@ -7,6 +7,7 @@ cassandra:
   endpointSnitch: "org.apache.cassandra.locator.Ec2Snitch"  # Snitch to be used in cassandra.yaml
                                                             # Needs to be changed in conjunction with multiRegionEnabled
   cassHome: "/etc/cassandra"                  # Path to the home dir of Cassandra
+  yamlLocation: "/etc/cassandra/conf/cassandra.yaml" # Path to Cassandra's YAML configuration
```
Comment (fahdsiddiqui) [inline priam/src/main/resources/conf/priam.yaml]: No need for this anymore. It will default to this location (see comment above).

**Comment by fahdsiddiqui** [comment]
@ToadJam - Other than a few comments.. LGTM :+1:

File: priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java (hunk 1 above)
Comment (ToadJam) [inline priam/src/main/java/com/netflix/priam/config/CassandraConfiguration.java]: @JsonIgnore seems to disable serialization/deserialization altogether, which removes the ability to override this value in situations where we need to (such as priam_local.yaml). Leaving this line as-is and adding the Objects.firstNonNull() logic you mention below seems to have the desired effect: using the YAML-specified "yamlLocation" when it exists, using the default value when it doesn't.

**Pull request merged**